mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -159,14 +157,21 @@ class GBTClassifier @Since("1.4.0") (
     val numFeatures = oldDataset.first().features.size
     val boostingStrategy = super.getOldBoostingStrategy(categoricalFeatures, OldAlgo.Classification)
 
-    val instr = Instrumentation.create(this, oldDataset)
+    val numClasses: Int = getNumClasses(dataset)
```

Comment:
We should just use numClasses = 2 for now, since `getNumClasses` can make an extra pass over the data, and >2 classes are not supported anyway.